Fleece/Core/Doc.cc

```diff
@@ -121,11 +136,14 @@ namespace fleece { namespace impl {
                 Log("Duplicate  (%p ... %p) --> Scope %p, sk=%p",
                     _data.buf, _data.end(), this, _sk.get());
             } else {
+                static const char* const valueTypeNames[] {"Null", "Boolean", "Number", "String", "Data", "Array", "Dict"};
+                auto type1 = ((const Value*)_data.buf)->type();
+                auto type2 = ((const Value*)existing->_data.buf)->type();
```

Comment:
This will not work -- the root Value in encoded Fleece data is (basically) at the end, not the beginning. It only happens to be at the beginning in the degenerate case when the size is 2.

If you want to get the root type you'd have to call `Value::fromData(_data)->type()`, which will parse the data.

Reply:
But it will get the right type if the length of the slice is 2, or not? Or perhaps I should try to get the type of general case.